### PR TITLE
Make sure we generate a proper docker tag

### PR DESCRIPTION
**Pull request by krnowak**
I have a branch named like <user>/<feature>, which becomes a part of
the tag name. Docker complains about it, because / is not a valid
character in the tag.

This commit replaces all sequences of invalid chars with a single dash
and makes sure that the tag name never exceeds the 127 chars limit.

### CODE REVIEW

**Comment by alban** [comment]
This is similar to https://github.com/weaveworks/scope/pull/1655

And probably failing for the same reason.

**Comment by tomwilkie** [comment]
Hi @krnowak; can you rebase this on top of master?  It should pass now.

**Comment by krnowak** [comment]
Updated, lint seems to fail with some typos. For lint fixes please see https://github.com/weaveworks/scope/pull/1751.

File: image-tag [hunk 1]
```diff
@@ -8,5 +8,10 @@ WORKING_SUFFIX=$(if ! git diff --exit-code --quiet HEAD >&2; \
                  then echo "-WIP"; \
                  else echo ""; \
                  fi)
-BRANCH_PREFIX=$(git rev-parse --abbrev-ref HEAD)
-echo "${BRANCH_PREFIX//\//-}-$(git rev-parse --short HEAD)$WORKING_SUFFIX"
```
Comment (alban) [inline image-tag]: For consistency, should `${CIRCLE_BRANCH//\//-}` be changed in `circle.yml` as well?

**Comment by krnowak** [comment]
Updated, so we can be sure that tag names in circleCI are also valid.

**Comment by 2opremio** [comment]
@krnowak / @alban Do we still need this?

**Comment by krnowak** [comment]
Up to you, really. But now I guess this PR would need to be split in two - one for image-tag to be filed in build-tools repository and one for scope importing the new build-tools version and modifying circle.yml.

My issue was the same as the one @alban had - I had / in my branch name and docker did not like it. I just made extra sure that the tag we generate based on a branch name is always valid (no invalid chars and shorter than 128 chars).

**Comment by tomwilkie** [comment]
Already an image-tag in build-tools: https://github.com/weaveworks/build-tools/blob/master/image-tag

**Pull request closed without merging**